actionpack/lib/action_dispatch/middleware/static.rb

```diff
@@ -82,7 +82,7 @@ def content_type(path)
       end
 
       def gzip_encoding_accepted?(request)
-        request.accept_encoding =~ /\bgzip\b/i
+        request.accept_encoding.any? { |enc, quality| enc =~ /\bgzip\b/i }
```

Comment:
This will now only work with `Rack::Request` objects. The only place that `FileHandler` gets called from is this file, so I think that's fine, but let me know if you think we should defend against receiving an `ActionDispatch::Request` here.